plugin/federation/federation.gotpl

```diff
@@ -310,10 +313,15 @@ func (ec *executionContext) resolveManyEntities(
 						{{- range $i, $field := .Field }}
 							val, ok = m["{{.}}"]
 							if !ok {
+								entityResolverErrs = append(entityResolverErrs,
+									fmt.Errorf("%w due to missing Key Field \"{{.}}\" for {{$entity.Name}}", ErrTypeNotFound))
 								break
 							}
 							{{- if (ne $i $keyField.Field.LastIndex ) }}
```

Comment:
I am not very sure I understand what this code is doing here.
We only want to possibly skip suggesting this resolver if the current field index does not match the last index of the key field?